src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -48,16 +49,115 @@ public async Task BestLocations_ReturnsOK(DateTimeOffset start, DateTimeOffset e
         _dataSourceMocker.SetupDataMock(start, end, location);
 
         //Call the private method to construct with parameters
-        var endpointURI = ConstructDateQueryURI(bestLocationsURI, location, start, end);
+        var queryStrings = new Dictionary<string, string>();
+        queryStrings["location"] = location;
+        queryStrings["time"] = $"{start:O}";
+        queryStrings["toTime"] = $"{end:O}";
+
+        var endpointURI = ConstructUriWithQueryString(bestLocationsURI,queryStrings);
 
         //Get response and response content
         var result = await _client.GetAsync(endpointURI);
 
         Assert.That(result, Is.Not.Null);
         Assert.That(result.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+    }
+
+    [TestCase("location", "", TestName = "empty location query string")]
+    [TestCase("non-location-param", "", TestName = "location param not present")]
+    public async Task BestLocations_EmptyLocationQueryString_ReturnsBadRequest(string queryString, string value)
```

Comment:
nit: Extra. It would be nice  to have covered from Integration Testing perspective a case that includes `['eastus', null]`. That would demonstrate unit and it coverage.

Reply:
I think for this PR we will just merge, but as we get into more detailed data creation I think we should add user input cases like this to the whole test suite.   To do so in the current state would require....yet another refactor

Reply:
I added it as a note on story 423 for follow-up